pandas/core/missing.py

```diff
@@ -94,6 +94,11 @@ def mask_missing(arr: ArrayLike, values_to_mask) -> npt.NDArray[np.bool_]:
             pass
         else:
             new_mask = arr == x
+            # GH#47480
+            if isinstance(new_mask, bool):
```

Comment:
Can you fix this without trying the failing conversion first? You can check for dtype object and if arr has any nans and use a different logic in that case.

I would suggest using another mask that applies the equality comparison only for values that are not nan